SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+//===-- ctor_fill.hpp - Functions for tests on simd fill constructor
+//      definition. -------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd fill constructor.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+// For std::isnan
+#include <cmath>
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+namespace esimd_tests = esimd_test::api::functional::ctors;
+namespace esimd_functional = esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_init =
+        esimd::simd<DataT, NumElems>(init_value, step);
+    simd_by_init.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in variable declaration
+// context.
+struct var_dec {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_var_decl(init_value, step);
```

Comment:
```suggestion
    const esimd::simd<DataT, NumElems> simd_by_var_decl(init_value, step);
```
Here and elsewhere: we might enforce const-correctness to state explicitly that this object is not going to change after the constructor was called. It may not defend from the `mutable` attributes in simd vector implementation (if any), but it definitely will trigger a failure in case of modyfying `copy_to()` call.
Minor for this specific case, still it may be a good practice to enforce const-correctness over the code in general.

Reply:
this comment was applied [here](https://github.com/intel/llvm-test-suite/pull/598/commits/652dcd3b7c72dceecf81bb7bd486398d8664c24d#diff-7fc90b589428e6e89076f07f5ff280a217b7f5c8e836bfe6c742734825074ef4R43-R44)

Reply:
Would you mind to apply both the `const-correctness` and the `auto` usage comments in every context descriptor used?

Reply:
Could you specify where `const-correctness` not applied?

Reply:
My bad, misread the code.